aws/base-settings/config.tf

```diff
@@ -0,0 +1,39 @@
+#=============================#
+# AWS Provider Settings       #
+#=============================#
+provider "aws" {
+  alias                   = "main_region"
+  region                  = var.region
+  profile                 = var.profile
+  shared_credentials_file = "~/.aws/${var.project}/config"
+}
+
+provider "aws" {
+  alias                   = "secondary_region"
+  region                  = var.region_secondary
+  profile                 = var.profile
+  shared_credentials_file = "~/.aws/${var.project}/config"
+}
+
+provider "vault" {
+  address = var.vault_address
+
+  /*
+  Vault token that will be used by Terraform to authenticate.
+  admin token from https://portal.cloud.hashicorp.com/.
+  */
+  token = var.vault_token
+}
+
+terraform {
+  required_version = ">= 0.14.2"
+
+  required_providers {
+    aws   = "~> 3.0"
+    vault = "~> 2.18.0"
+  }
+
+  backend "s3" {
+    key = "aws/base-settings/terraform.tfstate"
```

Comment:
@diego-ojeda-binbash Maybe to have a convention for the backend key we could have something like:
- vault-aws
- vault-hcp

Reply:
Yup, the key matches the relative path (aws/base-settings). Since the repository name already includes "vault" and since the repository is meant to hold Vault related configuration files, I figured there should be no need to also add "vault" in the backend key. What do you think?

Reply:
@diego-ojeda-binbash OJ agreed, so I should update the key of the hcp state to keep the convention here. Now is vault :( but I'll update it to hcp/...